### PR TITLE
[#4510] Disallow rodsusers irm -U'ing vault data (master)

### PR DESCRIPTION
**Pull request by alanking**
Needs run through CI.

### CODE REVIEW

**Comment by alanking** [comment]
Should probably add a few tests around `acSetChkFilePathPerm` to ensure that `rodsuser`s can unregister from resource vaults if that is the policy in place.

**Comment by alanking** [comment]
CI tests passed. iunreg (#4506) will handle tests related to `acSetChkFilePathPerm`.

File: scripts/irods/test/test_irm.py [hunk 1]
```diff
@@ -43,3 +44,37 @@ def test_irm_delete_collection_with_ampersand_in_name_causes_error__issue_3398(s
...
+            logical_path_to_obj = str(os.path.join(self.user.session_collection, filename))
+            self.user.assert_icommand(['irm', '-U', logical_path_to_obj], 'STDERR', 'CANT_UNREG_IN_VAULT_FILE')
+            self.user.assert_icommand(['ils', '-L', logical_path_to_obj], 'STDOUT', data_in_vault_path)
+            # Unregister from the vault as admin
+            self.admin.assert_icommand(['ichmod', '-M', 'own', self.admin.username, logical_path_to_obj])
+            self.admin.assert_icommand(['irm', '-U', logical_path_to_obj])
+            self.user.assert_icommand(['ils', '-L', logical_path_to_obj], 'STDERR', 'does not exist')
+            self.assertTrue(os.path.exists(data_in_vault_path), msg='Data missing after unregister in [{}]'.format(data_in_vault_path))
+
+            # Register a file in rodsuser's home collection and give write-only access
```
Comment (trel) [inline scripts/irods/test/test_irm.py]: write... isn't write-only.   it's 'read-write', but doesn't allow removal/unregistration.

**Comment by trel** [comment]
looks good - please to #

File: scripts/irods/test/test_irm.py (hunk 1 above)
Comment (alanking) [reply scripts/irods/test/test_irm.py]: I have updated the comment from "write-only" to "only read/write"

**Comment by alanking** [comment]
Applied #

**Pull request merged**